verifiers/internal/gha/bundle.go

```diff
@@ -156,6 +164,41 @@ func matchRekorEntryWithEnvelope(tlogEntry *v1.TransparencyLogEntry, env *dsseli
 	return nil
 }
 
+// matchRekorEntryWithEnvelopeDSSEv001 handles matchRekorEntryWithEnvelope for the dsse v0.0.1 type version.
+func matchRekorEntryWithEnvelopeDSSEv001(tlogEntry *v1.TransparencyLogEntry, env *dsselib.Envelope) error {
+	canonicalBody := tlogEntry.GetCanonicalizedBody()
+	var dsseObj models.DSSE
+	if err := json.Unmarshal(canonicalBody, &dsseObj); err != nil {
+		return fmt.Errorf("%w: %s", ErrorUnexpectedEntryType, err)
+	}
+	var dsseSchemaObj models.DSSEV001Schema
+	specMarshal, err := json.Marshal(dsseObj.Spec)
+	if err != nil {
+		return fmt.Errorf("%w: %s", ErrorUnexpectedEntryType, err)
+	}
+	if err := json.Unmarshal(specMarshal, &dsseSchemaObj); err != nil {
+		return fmt.Errorf("%w: %s", ErrorUnexpectedEntryType, err)
+	}
+	if len(env.Signatures) != len(dsseSchemaObj.Signatures) {
+		return fmt.Errorf("expected %d sigs in canonical body, got %d",
+			len(env.Signatures),
+			len(dsseSchemaObj.Signatures))
+	}
+	// TODO(#487): verify the certs match.
+	for _, sig := range env.Signatures {
+		var matchCanonical bool
+		for _, canonicalSig := range dsseSchemaObj.Signatures {
```

Comment:
can we use the new slices.contains functionality here? https://pkg.go.dev/slices#ContainsFunc

```
slices.ContainsFunc(dsseSchemaObj.Signatures, func(canonicalSig <type>) bool {
		canonicalSig.Signature == sig.Sig
	})
```

or something? I'm not super sure how this works.

Reply:
Added. It's not super clean, because I would have to reference the types of the inner objects, but it's a bit more readable, I think.